src/EditorFeatures/CSharpTest/InlineTypeCheck/CSharpInlineTypeCheckTests.cs

```diff
@@ -0,0 +1,367 @@
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (o is string x)
+        {
+        }
+        else
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        // prefix comment
+        [|var|] x = o as string;
+        if (x != null)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        // prefix comment
+        if (o is string x)
+        {
+        }
+    }
+}", compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|var|] x = o as string; // suffix comment
+        if (x != null)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        // suffix comment
+        if (o is string x)
+        {
+        }
+    }
+}", compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments3()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        // prefix comment
+        [|var|] x = o as string; // suffix comment
+        if (x != null)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        // prefix comment
+        // suffix comment
+        if (o is string x)
+        {
+        }
+    }
+}", compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheckComplexCondition1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|var|] x = o as string;
+        if (x != null ? 0 : 1)
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (o is string x ? 0 : 1)
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheckComplexCondition2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|var|] x = o as string;
+        if ((x != null))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if ((o is string x))
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheckComplexCondition3()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|var|] x = o as string;
+        if (x != null && x.Length > 0)
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (o is string x && x.Length > 0)
+        {
+        } 
+    }
+}");
+        }
```

Comment:
There are no tests for this due to where we landed on scoping.  Specifically, the scoping of the pattern variable is hte exact same as the scoping of the local variable declaration statement.  Since i only find and allow this when you have the local-declaration right next to the if-check, there is no concern.

Note that in my other PR (https://github.com/dotnet/roslyn/pull/14396) i do have tests for this because scoping *does* change (since hte local variable gets hoisted *outside* of the conditional check).